experiments/yarn/src/main/scala/org/apache/gearpump/experiments/yarn/master/ResourceManagerCallbackHandler.scala

```diff
@@ -60,19 +76,19 @@ class ResourceManagerCallbackHandler(appConfig: AppConfig, am: ActorRef) extends
 
     //todo:wont work with non uniform containers
     (0 until requestCount).foreach(request => {
-      am ! ContainerRequestMessage(containerCount, appConfig.getEnv(GEARPUMPMASTER_VCORES).toInt)
+      resourceManagerClient ! AdditionalContainersRequest(containerCount)
```

Comment:
I am not sure whether it is right to track the allocation status in this callback.
```scala
+  private val LOG: Logger = LogUtil.getLogger(getClass)
-  val allocatedContainersCount = new AtomicInteger(0)		+  private val completedContainersCount = new AtomicInteger(0)
-  val requestedContainersCount = new AtomicInteger(0)		+  private val failedContainersCount = new AtomicInteger(0)
+  private val allocatedContainersCount = new AtomicInteger(0)
+  private val requestedContainersCount = new AtomicInteger(0)
```

Should we decide how many additionaly container to allocate here by AdditionalContainersRequest or just forward the mesage ContianerAllocated to related parties.

With some minor modification, the code maybe greatly simplified. 
1. All message from ResourceManager extends a sealed trait like  ResourceManagerResponse.

2. Callback send the response to ResourceManagerClient, like RMError extends ResourceManagerResponse

3. In ResourceManagerClient, there is a receiver to handle ResourceManagerResponse, what it does is to forward that to YarnAppMaster.

4. The status like how many container are allocated, should be tracked by YarnAppMaster.